frontend/simple/js/event-log.js

```diff
@@ -0,0 +1,134 @@
+import localforage from 'localforage'
+import { toHash, makeEntry, makeGroup } from '../../../shared/functions'
+import request from 'superagent'
+
+// =======================
+// Helpers
+// =======================
+function getDB (hash) {
+  let db = localforage.createInstance({
+    name: 'Group Income',
+    storeName: hash
+  })
+  return db
+}
+function postEvent (groupId, hash, entry) {
+  return new Promise(function (resolve, reject) {
+    request.post(`${process.env.API_URL}/event/${groupId}`)
+      .send({hash, entry})
+      .end(function (err, res) {
+        if (err) {
+          return reject(err)
+        }
+        return resolve()
+      })
+  })
+}
+function postGroup (hash, entry) {
+  return new Promise(function (resolve, reject) {
+    request.post(`${process.env.API_URL}/group`)
+      .send({hash, entry})
+      .end(function (err, res) {
+        if (err) {
+          return reject(err)
+        }
+        return resolve(res.hash)
+      })
+  })
+}
+// =======================
+// Exports
+// =======================
+// get retrieves from the log
+// opts object allows for parent hash to be returned instead of data
+export async function get (log, opts) {
+  let hash
+  if (typeof opts === 'string') {
+    hash = opts
+  } else {
+    hash = opts.hash
+  }
+  let db = getDB(log.groupId)
+  let entry = await db.getItem(hash)
+  if (opts.parentHash) {
+    return entry.parentHash
+  } else {
+    return entry.data
+  }
+}
+// put appends to the log
+// opts allow for local updates only
+// returns the current state of the log
+export async function put (log, opts) {
+  let value
+  if (opts.value) {
+    value = opts.value
+  } else {
+    value = opts
+  }
```

Comment:
Same here. It is standard for a "put" type function to have the value that's being added not in the options, but in the parameters, so remove this `if`-block and change the function signature to something like:

```js
export async function addItemToLog(logId, item, opts) {

}
```